### DISCUSSION TITLE
How to mark optional query string?

### DISCUSSION THREAD

**Discussion by goni098** [juhaku/utoipa#683] [Q&A] [answered]
My query struct 
```
#[derive(Deserialize, IntoParams, Validate)]
pub struct QueryString {
  #[validate(range(min = 1))]
  limit: u32,

  r#type: Option<String>,

  main_category: Option<String>,

}
```
And on swagger ui type and main_category that are required. How to mark its is optional?

**Comment by juhaku** [comment, marked as answer]
Hey, this is related to issue https://github.com/juhaku/utoipa/issues/622. The properties should be nullable by default when `Option` is used, but it seems that there is some bug that causes properties to be required. 

Others has solved this with providing manually the `parameter_in = Query` attribute for the `IntoParams` macro.

**Reply by goni098** [reply]
thank you
